x-pack/plugins/fleet/server/services/package_policy.ts

```
@@ -1250,53 +1329,95 @@ class PackagePolicyClientImpl implements PackagePolicyClient {
+            let updatedNewData = newData;
+            for (const callback of externalCallbacks) {
+              let result;
+              if (externalCallbackType === 'packagePolicyPostCreate') {
+                result = await (callback as PostPackagePolicyPostCreateCallback)(
+                  updatedNewData as PackagePolicy,
+                  soClient,
+                  esClient,
+                  context,
+                  request
+                );
+                updatedNewData = PackagePolicySchema.validate(result);
+              } else {
+                result = await (callback as PostPackagePolicyCreateCallback)(
+                  updatedNewData as NewPackagePolicy,
+                  soClient,
+                  esClient,
+                  context,
+                  request
+                );
+              }
+
+              if (externalCallbackType === 'packagePolicyCreate') {
+                updatedNewData = NewPackagePolicySchema.validate(result);
+              } else if (externalCallbackType === 'packagePolicyUpdate') {
+                const omitted = {
+                  ...omit(result, [
+                    'id',
+                    'version',
+                    'revision',
+                    'updated_at',
+                    'updated_by',
+                    'created_at',
+                    'created_by',
+                    'elasticsearch',
+                  ]),
+                  inputs: result.inputs.map((input) => omit(input, ['compiled_input'])),
+                };
+
+                updatedNewData = UpdatePackagePolicySchema.validate(omitted);
+              }
             }
-          }
 
-          newData = updatedNewData;
+            newData = updatedNewData;
+          }
+          return newData;
         }
-        return newData;
       }
+    } catch (error) {
+      logger.error(`Error running external callbacks for ${externalCallbackType}:`);
+      logger.error(error);
+      throw error;
     }
```

I decided to re-throw the error but perhaps it's better to catch the error and not re-throw. This way we can remove all the try/catch blocks around the `packagePolicyService.runExternalCallbacks` invocations.

@kpollich @nchaulet WDYT?

---

I think we should probably keep the throw, but it seems we currently have a different behavior for update/create and delete. It seems create/update throw and we swallow error on delete

---

Yes i noticed the same inconsistency.